docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -65,6 +65,72 @@ This change affects almost all components where you're using the `component` pro
   +},
   ```
 
+- The components' definition inside the theme were restructure under the `components` key, to allow people easier discoverability about the definitions regarding one component.
+
+1. `props`
+
+```diff
+import { createMuiTheme } from '@material-ui/core/styles';
+
+const theme = createMuitheme({
+-  props: {
+-    MuiButton: {
+-      disableRipple: true,
+-    },
+-  },
++  components: {
++    MuiButton: {
++      props: {
++        disableRipple: true,
++      },
++    },
++  },
+});
+```
+
+2. `overrides`
+
+```diff
+import { createMuiTheme } from '@material-ui/core/styles';
+
+const theme = createMuitheme({
+-  overrides: {
+-    MuiButton: {
+-      root: { padding: 0 },
+-    },
+-  },
++  components: {
++    MuiButton: {
++      overrides: {
++        root: { padding: 0 },
++      },
++    },
++  },
+});
+```
+
+For allowing smoother transition, we've created an adapter for v4 theme that clients can use, so they can iteratively upgrade to the new theme structure.
```

Comment:
```suggestion
For a smoother transition, the `adaptV4Theme` helper allows you to iteratively upgrade to the new theme structure. Note that it will display a deprecation warning in the console, since it will be removed at the next major release.
```